src/test/java/htsjdk/samtools/SAMRecordUnitTest.java

```diff
@@ -1039,4 +1039,20 @@ public SAMRecord createTestSamRec() {
 
         return(rec);
     }
+
+    @DataProvider
+    public Object [][] readBasesGetReadLengthData() {
```

Comment:
Can you also add tests that set the bases using `setReadString`? in particular try to break it with an empty string and the "*" string that means the empty string for read bases.